src/allmydata/test/cli/test_create.py

```diff
@@ -287,24 +295,19 @@ def test_node_basedir_exists(self):
         self.assertIn("To avoid clobbering anything, I am going to quit now", err)
 
     @defer.inlineCallbacks
-    def test_node_slow_tor(self):
-        basedir = self.mktemp()
+    def test_node_slow(self):
+        """
+        A node can be created using a listener type that returns an
+        unfired Deferred from its ``create_config`` method.
+        """
         d = defer.Deferred()
-        self.patch(tor_provider, "create_config", lambda *a, **kw: d)
-        d2 = run_cli("create-node", "--listen=tor", basedir)
-        d.callback(({}, "port", "location"))
-        rc, out, err = yield d2
-        self.assertEqual(rc, 0)
-        self.assertIn("Node created", out)
-        self.assertEqual(err, "")
+        slow = StaticProvider(True, False, d, None)
+        create_node._LISTENERS["xxyzy"] = slow
+        self.addCleanup(lambda: create_node._LISTENERS.pop("xxyzy"))
```

Comment:
Is this going to break parallel tests, or does trial use subprocesses?

Reply:
`--jobs N` launches N child processes and in each child runs tests sequentially, yea, so this kind of global state mutation is not immediately catastrophic.

I thought about trying to parameterize `_LISTENERS` instead of having the global but the users are hidden deep in the call stack :/    So I didn't.